### ISSUE TITLE
Block buying/selling on specific account(s) for specified brokerage 

### ISSUE THREAD

**Issue by Flouzr** [NelsonDane/auto-rsa#179]
Maybe I missed it when reading the documentation, but is it possible to block a specific account when buying/selling?

The reason I ask is I have a Roth on Fidelity that I would like to not buy/sell on and I currently do not see a way to have that account skipped.

This would probably also be useful for other brokerages too.

**Comment by NelsonDane** [comment]
I agree and this is something I've been discussing/pondering. I'm still trying to figure out the best way to input this in the command line in a way that makes sense tho

**Comment by Flouzr** [comment]
Some initial thoughts on my end.

- For every supported brokerage have a `<BROKERAGE>_EXCLUDED_ACCOUNTS` field that is a comma separated list. Ex: `FIDELITY_EXCLUDED_ACCOUNTS: Z0000001, Z0000002`
- This list is read, kept in runtime, and checked when iterating through each account per brokerage (as not to read the env every time)
- If the account ID matches, then skip (and possibly print that the account was skipped for user feedback)

If I get time I'll see if I can get a PR together but I just came back from a trip so I'll be playing catch-up at work for a good bit...

**Comment by jwhitefield42** [comment]
> Some initial thoughts on my end.
> 
> * For every supported brokerage have a `<BROKERAGE>_EXCLUDED_ACCOUNTS` field that is a comma separated list. Ex: `FIDELITY_EXCLUDED_ACCOUNTS: Z0000001, Z0000002`
> * This list is read, kept in runtime, and checked when iterating through each account per brokerage (as not to read the env every time)
> * If the account ID matches, then skip (and possibly print that the account was skipped for user feedback)
> 
> If I get time I'll see if I can get a PR together but I just came back from a trip so I'll be playing catch-up at work for a good bit...

Down side of this is if a user only wants to exclude an account sometimes (e.g. for PDT protection). A CLI flag to skip would allow this to be optional. Maybe an alternative is to have a skip account field in the ENV, that instead of skipping, prompts the user whether to buy or not...

**Comment by NelsonDane** [comment]
> Some initial thoughts on my end.
> 
> 
> 
> - For every supported brokerage have a `<BROKERAGE>_EXCLUDED_ACCOUNTS` field that is a comma separated list. Ex: `FIDELITY_EXCLUDED_ACCOUNTS: Z0000001, Z0000002`
> 
> - This list is read, kept in runtime, and checked when iterating through each account per brokerage (as not to read the env every time)
> 
> - If the account ID matches, then skip (and possibly print that the account was skipped for user feedback)
> 
> 
> 
> If I get time I'll see if I can get a PR together but I just came back from a trip so I'll be playing catch-up at work for a good bit...

This is a good thought if you wanted to permanently exclude accounts. However if you just want to occasionally exclude a single account, this wouldn't be a good option. Especially for the discord bot where editing the env file isn't as convenient
